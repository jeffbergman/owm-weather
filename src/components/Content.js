import React, { Component } from 'react';
import { connect } from 'react-redux';
import Current from './Current';
import Forecast from './Forecast';
import { fetchWeather, noGeo, geoError } from '../actions';

class Content extends Component {
	constructor(props) {
		super(props);
		this.geoLocate = this.geoLocate.bind(this);
		this.geoSuccess = this.geoSuccess.bind(this);
		this.geoErr = this.geoErr.bind(this);
	}

  componentDidMount() {
  	//get new data based on the geolocation when the component mounts
    this.geoLocate();
  }

  geoLocate() {
    if (!navigator.geolocation){
      this.props.noGeo();  
      return;
    }
    navigator.geolocation.getCurrentPosition(this.geoSuccess, this.geoErr);
  }

  //these next 2 functions call the action creators 
  //that fetch the weather info or process the error 
  //from the geolocation
  //This code is duplicated in the search, 
  //but the alternative to this small duplication seems to be 
  //passing the action creators into functions in helpers.js,
  //which didn't seem right 
  geoSuccess(position) {
    this.props.fetchWeather(position);
  }

  geoErr() {
    this.props.geoError();  
  }

  render() {

  	const current = this.props.weather.current;
  	const forecast = this.props.weather.forecast || [];
  	const display = this.props.weather.display

  	//map the array of forecast data, only 4 items 
  	const forecasts = forecast.map(item => {
  		return (
				<Forecast key={item.day} item={item} display={display}/>
			);
  	})

  	return (
  		<div className="outer">
        <div className="cont">
        	<Current current={current}/>
        	{forecasts}
        </div>
      </div>
  	);
  }
}

const mapStateToProps = (state) => {
  return {
    error: state.error,
    weather: state.weather
  }
};

export default connect(mapStateToProps, { fetchWeather, noGeo, geoError })(Content);