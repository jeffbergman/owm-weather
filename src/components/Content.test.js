import React from 'react';
import { mount, shallow } from 'enzyme';
import Content from './Content';

  const mockFetchWeather = jest.fn();
  const mockUseGeoLocation = jest.fn();
  const mockNoGeo = jest.fn();
  const mockGeoError = jest.fn();
  const mockGeoLocate = jest.fn();

  let props = { 
    fetchWeather: mockFetchWeather,
    useGeoLocation: mockUseGeoLocation,
    noGeo: mockNoGeo,
    geoError: mockGeoError,
    weather: {
      current: { 
        "weather":[
          {"id":721,"main":"Haze","description":"haze","icon":"50d"},
          {"id":701,"main":"Mist","description":"mist","icon":"50d"}
        ],
        "main": {
          "temp":65.84,
          "pressure":1019,
          "humidity":43,
          "temp_min":59,
          "temp_max":69.8
        },
        "dt":1522346340,
        "sunrise":1522331019,
        "sunset":1522375970
      }  
    } 
  };

describe('Content', () => {

  let content = shallow(<Content.WrappedComponent {...props}/>);

  it('renders properly', () => {
    expect(content).toMatchSnapshot();
  });

});