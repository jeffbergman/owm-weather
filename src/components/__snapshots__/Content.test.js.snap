// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content
    fetchWeather={[Function]}
    geoError={[Function]}
    noGeo={[Function]}
    useGeoLocation={[Function]}
    weather={
        Object {
            "current": Object {
              "dt": 1522346340,
              "main": Object {
                "humidity": 43,
                "pressure": 1019,
                "temp": 65.84,
                "temp_max": 69.8,
                "temp_min": 59,
              },
              "sunrise": 1522331019,
              "sunset": 1522375970,
              "weather": Array [
                Object {
                  "description": "haze",
                  "icon": "50d",
                  "id": 721,
                  "main": "Haze",
                },
                Object {
                  "description": "mist",
                  "icon": "50d",
                  "id": 701,
                  "main": "Mist",
                },
              ],
            },
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="cont"
>
        <Current
                current={
                        Object {
                                "dt": 1522346340,
                                "main": Object {
                                  "humidity": 43,
                                  "pressure": 1019,
                                  "temp": 65.84,
                                  "temp_max": 69.8,
                                  "temp_min": 59,
                                },
                                "sunrise": 1522331019,
                                "sunset": 1522375970,
                                "weather": Array [
                                  Object {
                                    "description": "haze",
                                    "icon": "50d",
                                    "id": 721,
                                    "main": "Haze",
                                  },
                                  Object {
                                    "description": "mist",
                                    "icon": "50d",
                                    "id": 701,
                                    "main": "Mist",
                                  },
                                ],
                              }
                }
        />
</div>,
      "className": "outer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Current
            current={
                        Object {
                                    "dt": 1522346340,
                                    "main": Object {
                                      "humidity": 43,
                                      "pressure": 1019,
                                      "temp": 65.84,
                                      "temp_max": 69.8,
                                      "temp_min": 59,
                                    },
                                    "sunrise": 1522331019,
                                    "sunset": 1522375970,
                                    "weather": Array [
                                      Object {
                                        "description": "haze",
                                        "icon": "50d",
                                        "id": 721,
                                        "main": "Haze",
                                      },
                                      Object {
                                        "description": "mist",
                                        "icon": "50d",
                                        "id": 701,
                                        "main": "Mist",
                                      },
                                    ],
                                  }
            }
/>,
          Array [],
        ],
        "className": "cont",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "current": Object {
              "dt": 1522346340,
              "main": Object {
                "humidity": 43,
                "pressure": 1019,
                "temp": 65.84,
                "temp_max": 69.8,
                "temp_min": 59,
              },
              "sunrise": 1522331019,
              "sunset": 1522375970,
              "weather": Array [
                Object {
                  "description": "haze",
                  "icon": "50d",
                  "id": 721,
                  "main": "Haze",
                },
                Object {
                  "description": "mist",
                  "icon": "50d",
                  "id": 701,
                  "main": "Mist",
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="cont"
>
          <Current
                    current={
                              Object {
                                        "dt": 1522346340,
                                        "main": Object {
                                          "humidity": 43,
                                          "pressure": 1019,
                                          "temp": 65.84,
                                          "temp_max": 69.8,
                                          "temp_min": 59,
                                        },
                                        "sunrise": 1522331019,
                                        "sunset": 1522375970,
                                        "weather": Array [
                                          Object {
                                            "description": "haze",
                                            "icon": "50d",
                                            "id": 721,
                                            "main": "Haze",
                                          },
                                          Object {
                                            "description": "mist",
                                            "icon": "50d",
                                            "id": 701,
                                            "main": "Mist",
                                          },
                                        ],
                                      }
                    }
          />
</div>,
        "className": "outer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Current
              current={
                            Object {
                                          "dt": 1522346340,
                                          "main": Object {
                                            "humidity": 43,
                                            "pressure": 1019,
                                            "temp": 65.84,
                                            "temp_max": 69.8,
                                            "temp_min": 59,
                                          },
                                          "sunrise": 1522331019,
                                          "sunset": 1522375970,
                                          "weather": Array [
                                            Object {
                                              "description": "haze",
                                              "icon": "50d",
                                              "id": 721,
                                              "main": "Haze",
                                            },
                                            Object {
                                              "description": "mist",
                                              "icon": "50d",
                                              "id": 701,
                                              "main": "Mist",
                                            },
                                          ],
                                        }
              }
/>,
            Array [],
          ],
          "className": "cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "current": Object {
                "dt": 1522346340,
                "main": Object {
                  "humidity": 43,
                  "pressure": 1019,
                  "temp": 65.84,
                  "temp_max": 69.8,
                  "temp_min": 59,
                },
                "sunrise": 1522331019,
                "sunset": 1522375970,
                "weather": Array [
                  Object {
                    "description": "haze",
                    "icon": "50d",
                    "id": 721,
                    "main": "Haze",
                  },
                  Object {
                    "description": "mist",
                    "icon": "50d",
                    "id": 701,
                    "main": "Mist",
                  },
                ],
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
